neonvm/tools/vm-builder/files/resize-swap.sh

```diff
@@ -0,0 +1,33 @@
+#!/neonvm/bin/sh
+
+set -euo pipefail
+
+USAGE="USAGE: /neonvm/bin/resize-swap [ --once ] <NEW_SIZE_IN_BYTES>"
+
+if [ "$#" -eq 2 ]; then
+    if [ "$1" != '--once' ]; then
+        echo "expected first arg to be '--once'" >&2
+        echo "$USAGE" >&2
+        exit 1
+    fi
+    size="$2"
+    once=yes
+elif [ "$#" -eq 1 ]; then
+    size="$1"
+    once=no
+else
+    echo "expected 1 or 2 args, got $#" >&2
+    echo "$USAGE" >&2
+    exit 1
+fi
+
+if [ ! -f /neonvm/runtime/resize-swap.sh ]; then
```

Comment:
Why the need for two `resize-swap.sh`?

Reply:
> The short version is that one is the public interface, guaranteed to be stable, and the other is the implementation, which may change with new neonvm-runner versions.

So `/neonvm/bin/resize-swap` is stable, but `/neonvm/runtime/resize-swap.sh` is implementation-defined?

But anyway we control both scripts, so if the need to extend capability arises, we can change it in a backward-compatible way (or introduce new script). WDYT?

Reply:
We could in theory, yeah. In practice, maintaining backwards compatibility is hard IMO — I think the contract is more explicit if it's very clearly split across components.

And also, we only currently inject into `/neonvm/runtime` and not `/neonvm/bin`, so either compute_ctl would have to know about `/neonvm/runtime`, which feels like relying on implementation details IMO, or we set up _something_ to link between them (which is relatively close to what we have anyways). And it's easier for that to be a separate script than a symlink, because inserting a new executable file into `/neonvm/runtime` requires fiddling with the file permissions, which we currently get to ignore.

Does that make sense? wdyt?

Reply:
> Does that make sense?

TBH, still not fully. I don't see the problem neither with runner injecting into `/neonvm/bin`, not with teaching `compute_ctl` to call `/neonvm/runtime/resize-swap`. We have only one implementation of neonvm-runner, so the notion of "implementation-defined" doesn't make much sense. I do acknowledge one file is tied to the `neonvm-runner` version, and the other to the compute image version, but I don't like this distinction, and don't think it should dictate any design choices which have no practical implications.

Probably you see some abstraction layer which I don't see.

Also, if we keep both scripts one should be renamed, e.g.  `/neonvm/runtime/resize-swap-internal.sh`. Every time it takes me some time to figure out which script calls which.